core/src/SYCL/Kokkos_SYCL_ParallelReduce_MDRange.hpp

```diff
@@ -95,6 +95,11 @@ class Kokkos::Impl::ParallelReduce<CombinedFunctorReducerType,
     const unsigned int value_count =
         m_functor_reducer.get_reducer().value_count();
     sycl::device_ptr<value_type> results_ptr;
+    auto host_result_ptr =
+        (m_result_ptr && !m_result_ptr_device_accessible)
+            ? static_cast<sycl::host_ptr<value_type>>(
+                  instance.scratch_host(sizeof(value_type) * value_count))
```

Comment:
Thread safety comes from the shared memory mutex?
If you don't rename it, at least add a comment that it also guards for the host-pinned buffer.

Reply:
2ea1f0e16da449b714962ce331086b61afadecf2